### PR TITLE
Add a mock server for tests [develop]

### PR DESCRIPTION
**Pull request by Samweli**
Based on https://github.com/ConservationInternational/trends.earth/pull/742, 

Adds the following changes that are related to the tests.

- Updates on the `api` module now using a `APIClient` class to store and interface all the API related functionality this makes the module testable.
- Removal of unused methods in `calculate.py`
- Added a new mock API server for running tests local and on CI
- Include initial test for the `calculate_ldn` alg.

### CODE REVIEW

**Comment by azvoleff** [comment]
I get an immediate crash if I try to run this branch in QGIS, so I'm hesitant to merge at this point as I it would complicate testing any other PRs relative to develop. 

I don't get a crash report when it crashes, but I do see this in the log pane (looks like a missing method):

image

Are you able to run this on your machine?

**Comment by Samweli** [comment]
> I don't get a crash report when it crashes, but I do see this in the log pane before QGIS closes (looks like a missing method):

@azvoleff same thing happened to me, I have made a fix for it here https://github.com/ConservationInternational/trends.earth/pull/745/commits/60f5ae6ce64e53bc3ce04da860d56e74b47c095a

File: LDMP/api.py [hunk 1]
```diff
@@ -126,324 +144,339 @@ def finished(self, result):
...
-            )
-            """
             ret = None
-    else:
-        ret = None
 
-    return ret
+        return ret
 
+    def get_header(self, url):
```
Comment (azvoleff) [inline LDMP/api.py:360]: This function doesn't actually use the API, so would make more sense to handle this in a different class (it is for getting the header of any generic URL, needed for checking file size, etc.)

**Comment by azvoleff** [comment]
Still hesitant to merge this one @Samweli. It seems to break task submission - when you try to submit a remote task it appears nothing happens (the job doesn't show up on our server nor in the plugin, and there are no error messages shown to the user). It seems to be an issue with getting the token.

**Comment by Samweli** [comment]
> when you try to submit a remote task it appears nothing happens

@azvoleff this was indeed an issues in getting the `token`, I have fixed and tested it see the below screenshot
image

**Comment by azvoleff** [comment]
Thanks - looks good~

**Pull request merged**